### PR TITLE
Added indeterminate feature

### PR DESCRIPTION
**Pull request by alex-e-leon**
Added the indeterminate prop to enhancedSwitch which adds the
appropriate indeterminate class and then attempts to set the input to
indeterminate onMount and onUpdate.
See https://github.com/facebook/react/issues/1798 for discussion about setting indeterminate with react.

### CODE REVIEW

**Comment by luqin** [comment]
CI checks have failed

**Comment by alex-e-leon** [comment]
Ran the linter and tests and fixed it.

File: lib/EnhancedSwitch.js [hunk 1]
```diff
@@ -247,9 +247,11 @@ var EnhancedSwitch = (function (_React$Component) {
       }
     }
   }, {
-    key: 'isChecked',
-    value: function isChecked() {
-      return this.refs.checkbox.checked;
+    key: 'setIndeterminate',
+    value: function setIndeterminate() {
+      if (this.props.indeterminate) {
+        this.refs.checkbox.indetermiante = true;
```
Comment (dclowd9901) [inline lib/EnhancedSwitch.js:253]: "indetermiante"? This is broken -- there is no "indetermiante" property on dom checkbox elements. Now, "indeterminate" is a different story...

Comment (alex-e-leon) [inline lib/EnhancedSwitch.js:253]: @dclowd9901 Good catch. This is what happens when you have to write `indeterminate` way too may times. Created a PR to fix it

Comment (dclowd9901) [inline lib/EnhancedSwitch.js:253]: Happy to help :) using a self made polyfill now, so I'm def looking forward to this.

**Pull request merged**